examples/paho-mqtt/main.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (C) 2019 Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file        main.c
+ * @brief       Example using MQTT Paho package from RIOT
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include "xtimer.h"
+#include "shell.h"
+#include "thread.h"
+#include "mutex.h"
+#include "mqtt.h"
+#include "MQTTClient.h"
+
+#define BUF_SIZE                        1024
+#define MQTT_VERSION_v311               4       /* MQTT v3.1.1 version is 4 */
+#define COMMAND_TIMEOUT_MS              4000
+
+#define DEFAULT_MQTT_PORT               1883    /**< Default MQTT port */
+#define DEFAULT_KEEPALIVE_SEC           10      /**< Keepalive timeout in seconds */
+#define DEFAULT_MQTT_USER               "riot"
+#define DEFAULT_MQTT_PWD                 ""
+
+#ifndef MAX_LEN_TOPIC
+#define MAX_LEN_TOPIC                   100
+#endif
+
+#ifndef MAX_TOPICS
+#define MAX_TOPICS                      4
+#endif
+
+#define IS_CLEAN_SESSION                1
+#define IS_RETAINED_MSG                 0
+
+static MQTTClient client;
+static Network network;
+static int topic_cnt = 0;
+static char _topic_to_subscribe[MAX_TOPICS][MAX_LEN_TOPIC];
+
+static unsigned get_qos(const char *str)
+{
+    int qos = atoi(str);
+
+    switch (qos) {
+    case 1:     return QOS1;
+    case 2:     return QOS2;
+    default:    return QOS0;
+    }
+}
+
+static void _on_msg_received(MessageData *data)
+{
+    printf("paho_mqtt_example: message received on topic"
+           " %.*s: %.*s\n",
+           (int)data->topicName->lenstring.len,
+           data->topicName->lenstring.data, (int)data->message->payloadlen,
+           (char *)data->message->payload);
+}
+
+static int _cmd_discon(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    topic_cnt = 0;
+    int res = MQTTDisconnect(&client);
+    if (res < 0) {
+        printf("mqtt_example: Unable to disconnect\n");
+    }
+    else {
+        printf("mqtt_example: Disconnect successful\n");
+    }
+
+    NetworkDisconnect(&network);
+    return res;
+}
+
+static int _cmd_con(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf(
+            "usage: %s <ipv6 addr> [user] [password] [port] [keepalivetime] \n",
```

Comment:
I think `port` should the first optional argument after the address. IMHO, it makes no sense to have it after user credentials. I also think that sharing the username and the clientid is a bad idea. There should 2 different optional parameters for them.
Is it possible to connect with an empty clientid ? If not use an overridable define that is "riot" by default. E.g. after the default user/password defines, add:
```
#ifndef DEFAULT_MQTT_CLIENTID
#define DEFAULT_MQTT_CLIENTID "riot"
#endif
```